app/services/payment_provider_customers/gocardless_service.rb

```diff
@@ -25,17 +25,18 @@ def create
       result
     end
 
-    def generate_checkout_url
+    def generate_checkout_url(send_webhook: true)
       billing_request = create_billing_request(gocardless_customer.provider_customer_id)
       billing_request_flow = create_billing_request_flow(billing_request.id)
 
+      result.checkout_url = billing_request_flow.authorisation_url
+
       SendWebhookJob.perform_later(
         'customer.checkout_url_generated',
         customer,
-        checkout_url: billing_request_flow.authorisation_url,
-      )
+        checkout_url: result.checkout_url,
```

Comment:
Style/MultilineIfModifier: Favor a normal if-statement over a modifier clause in a multiline statement. (https://rubystyle.guide#no-multiline-if-modifiers)